pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go

```diff
@@ -273,6 +278,36 @@ func (c *Controller) syncEndpoints(key string) error {
 		return err
 	}
 
+	endpoints, err := c.endpointsLister.Endpoints(namespace).Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return c.deleteMirroredSlices(namespace, name)
+		}
+		return err
+	}
+
+	if !c.shouldMirror(endpoints) {
+		return c.deleteMirroredSlices(namespace, name)
+	}
+
+	svc, err := c.serviceLister.Services(namespace).Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof("Skipping mirroring for %s/%s due to missing Service", namespace, name)
+			return c.deleteMirroredSlices(namespace, name)
+		}
+		klog.Errorf("Error fetching %s/%s Service: %v", namespace, name, err)
+		return err
+	}
+
+	// This means that if a Service transitions away from a nil selector, any
+	// mirrored EndpointSlices will not be cleaned up. This matches the behavior
+	// of the Endpoints and EndpointSlice controllers.
+	if svc.Spec.Selector != nil {
+		klog.V(4).Infof("Not mirroring EndpointSlice for %s/%s because selector is not nil", namespace, name)
```

Comment:
won't this get logged a *ton* (basically on every endpoints update for almost all services)? I'm not sure I'd log here at all

Reply:
Good call, removed.